Scripts/Runtime/Data/LazyReactiveValue.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using Anvil.CSharp.Core;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    /// <summary>
+    /// A wrapper for a value that checks a trigger source on each get and updates its value before returning the value.
+    /// </summary>
+    /// <typeparam name="TValue">The type of the value</typeparam>
+    /// <typeparam name="TTriggerType">The type of the trigger source's value</typeparam>
+    /// <example>
+    /// The most common use case for this type is to lazily recompute or cache an <see cref="AccessControlledValue"/>. The below
+    /// example is contrived but demonstrates usage. Typically the data fetched from the AccessControlledValue would be
+    /// transformed in a way that would make caching worthwhile.
+    /// <code>
+    /// AccessControlledValue<int> myACV = new AccessControlledValue<int>(5);
+    /// LazyReactiveValue<uint, JobHandle> myCachedValue = new LazyReactiveValue(
+    ///     () => myACV.GetDependencyFor(AccessType.SharedRead),
+    ///     (ref uint value) =>
+    ///     {
+    ///         using var myACVHandle = myACV.AcquireWithHandle(AccessType.SharedRead);
+    ///         value = myACVHandle.Value;
+    ///     }
+    /// );
+    /// </code>
+    /// </example>
+    public class LazyReactiveValue<TValue, TTriggerType> : AbstractAnvilBase
+    {
+        /// <summary>
+        /// A delegate that is expected to modify the <see cref="value"/> parameter in place with up to date value(s).
+        /// </summary>
+        /// <param name="value">A reference to the value to update.</param>
+        public delegate void UpdateValueDelegate(ref TValue value);
+
+        private readonly Func<TTriggerType> m_GetCurrentTriggerValue;
+        private readonly UpdateValueDelegate m_UpdateValue;
+
+        private TValue m_Value;
+        private TTriggerType m_LastDependencyHandle;
```

Comment:
Minor but since this isn't a handle anymore, rename? 
```suggestion
        private TTriggerType m_LastTrigger;
```

Reply:
Fixed